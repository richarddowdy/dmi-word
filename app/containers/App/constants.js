/*
 * AppConstants
 */

export const ADD_STRINGS_SUCCESS = 'word-dmi/Form/ADD_STRINGS_SUCCESS';
export const ADD_STRINGS_FAIL = 'word-dmi/Form/ADD_STRINGS_FAIL';
