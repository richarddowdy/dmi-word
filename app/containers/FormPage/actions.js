/*
 *
 * Form actions
 *
 */

import {
  CHANGE_INPUT,
  ADD_NEW_STRING,
  LOAD_STRINGS_SUCCESS,
  ADD_STRINGS_SUCCESS,
  ADD_STRINGS_FAIL,
} from './constants';

export function changeInput(newString) {
  return {
    type: CHANGE_INPUT,
    newString,
  };
}

export function addNewString(newString) {
  return {
    type: ADD_NEW_STRING,
    newString,
  };
}

export function stringAdd(stringsArray) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    stringsArray,
  };
}

export function stringFailed(error) {
  return {
    type: ADD_STRINGS_FAIL,
    error,
  };
}

export function stringSuccess(success) {
  return {
    type: ADD_STRINGS_SUCCESS,
    success,
  };
}
