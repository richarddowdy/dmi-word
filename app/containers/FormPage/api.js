import axios from 'axios';

const BASE_URL = 'http://localhost:3000/api/strings';

async function addNewStringsApi(string) {
  try {
    const response = await axios.post(BASE_URL, { newString: string })
    return response.data;
  } catch (err) {
    return err;
  }
}

export { addNewStringsApi };
