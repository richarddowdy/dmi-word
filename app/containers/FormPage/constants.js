/*
 *
 * Form constants
 *
 */

export const CHANGE_INPUT = 'word-dmi/Form/CHANGE_INPUT';
export const ADD_NEW_STRING = 'word-dmi/Form/ADD_NEW_STRING';
export const LOAD_STRINGS_SUCCESS = 'word-dmi/Form/LOAD_STRINGS_SUCCESS';
