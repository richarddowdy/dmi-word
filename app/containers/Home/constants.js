/*
 *
 * Home constants
 *
 */

export const DEFAULT_ACTION = 'app/Home/DEFAULT_ACTION';
export const REQUEST_API_STRINGS = 'word-dmi/Home/REQUEST_API_STRINGS';
export const RECEIVE_API_STRINGS = 'word-dmi/Home/RECEIVE_API_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'word-dmi/Home/LOAD_STRINGS_SUCCESS';
export const DELETE_STRING = 'word-dmi/Home/DELETE_STRING';
export const DELETE_SUCCESS = 'word-dmi/Home/DELETE_SUCCESS';
export const DELETE_FAIL = 'word-dmi/Home/DELETE_FAIL';
export const RESET_SUCCESS = 'word-dmi/Home/RESET_SUCCESS';
