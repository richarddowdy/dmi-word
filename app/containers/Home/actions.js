/*
 *
 * Home actions
 *
 */

import { DEFAULT_ACTION, REQUEST_API_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export function loadStrings() {
  return {
    type: REQUEST_API_STRINGS,
  };
}

export function stringsLoaded(strings) {
  console.log("pushing to redux", strings)
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}
