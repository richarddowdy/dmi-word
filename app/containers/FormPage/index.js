/**
 *
 * Form
 *
 */

import React, { memo, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { useInjectSaga } from 'utils/injectSaga';
import { useInjectReducer } from 'utils/injectReducer';
import { makeSelectForm, makeSelectError, makeSelectSuccess } from './selectors';
import reducer from './reducer';
import saga from './saga';

import { changeInput, addNewString, stringSuccess, stringFailed } from './actions';

import Form from './Form';
import Input from './Input';
import Alert from '../../components/Alert/index';

export function FormPage({ newString, handleChange, handleSubmit, error, success, resetSuccess, resetError }) {
  useInjectReducer({ key: 'form', reducer });
  useInjectSaga({ key: 'form', saga });

  useEffect(() => {
    if (success) {
      resetSuccess();
    }
    if (error) {
      resetError();
    }
  }, []);

  return (
    <div>
      <Helmet>
        <title>Form</title>
        <meta name="description" content="Add word or phrase to list." />
      </Helmet>
      <Form onSubmit={handleSubmit}>
        <label htmlFor="newString">Add a New Word or Phrase to the list!</label>
        <Input
          className="form-group"
          id="new-string"
          type="text"
          placeholder="New Phrase!"
          value={newString}
          onChange={handleChange}
          onClick={success ? resetSuccess : error ? resetError : null}
        />
        <button type="submit" className="btn btn-primary">Submit</button>
      </Form>
      {error ? <Alert className="alert alert-danger">{error}</Alert> : null}
      {success ? <Alert className="alert alert-success">{success}</Alert> : null}
    </div>
  );
}

FormPage.propTypes = {
  newString: PropTypes.string,
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  success: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
};

const mapStateToProps = createStructuredSelector({
  newString: makeSelectForm(),
  error: makeSelectError(),
  success: makeSelectSuccess(),
});

function mapDispatchToProps(dispatch) {
  return {
    handleChange: evt => dispatch(changeInput(evt.target.value)),
    handleSubmit: evt => {
      evt.preventDefault();
      dispatch(addNewString());
      dispatch(changeInput(''));
    },
    resetSuccess: () => dispatch(stringSuccess(false)),
    resetError: () => dispatch(stringFailed(false)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(FormPage);
