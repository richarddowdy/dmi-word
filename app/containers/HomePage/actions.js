/*
 *
 * Home actions
 *
 */

import {
  REQUEST_API_STRINGS,
  LOAD_STRINGS_SUCCESS,
  DELETE_STRING,
  DELETE_FAIL,
  DELETE_SUCCESS,
} from './constants';

import { LOAD_STRINGS_FAIL } from '../FormPage/constants';

export function loadStrings() {
  return {
    type: REQUEST_API_STRINGS,
  };
}

export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

export function stringsFailed(error) {
  return {
    type: LOAD_STRINGS_FAIL,
    error,
  };
}

export function deleteString(string) {
  return {
    type: DELETE_STRING,
    string,
  };
}

export function deleteFailed(error) {
  return {
    type: DELETE_FAIL,
    error,
  };
}

export function deleteSuccess(strings) {
  return {
    type: DELETE_SUCCESS,
    strings,
  };
}
