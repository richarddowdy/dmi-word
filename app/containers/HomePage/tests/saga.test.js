/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { take, call, put, select } from 'redux-saga/effects';
import homeSaga, { getApiStrings } from '../saga';
import { REQUEST_API_STRINGS, DELETE_STRING, LOAD_STRINGS_SUCCESS } from '../constants';
import { stringsLoaded } from '../actions';
import { stringFailed } from '../../FormPage/actions';

// const generator = homeSaga();

// describe('homeSaga Saga', () => {
//   it('Expect to have unit tests specified', () => {
//     expect(true).toEqual(false);
//   });
// });

const strings = ['one', 'two', 'three'];

describe('getApiStrings Saga', () => {
  let getApiStringsGenerator;

  beforeEach(() => {
    getApiStringsGenerator = getApiStrings();

    const selectDescriptor = getApiStringsGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getApiStringsGenerator.next(strings).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should call the stringFailed action if the response errors', () => {
    const response = new Error('Error');
    const putDescriptor = getApiStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(stringFailed(response)));
  });
});
