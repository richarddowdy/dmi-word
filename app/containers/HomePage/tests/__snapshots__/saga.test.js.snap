// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getApiStrings Saga should call the stringFailed action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getApiStrings Saga should call the stringFailed action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "strings": Array [
        "one",
        "two",
        "three",
      ],
      "type": "word-dmi/Home/LOAD_STRINGS_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
